components/server/ee/src/user/user-deletion-service.ts

```diff
@@ -61,7 +61,7 @@ export class UserDeletionServiceEE extends UserDeletionService {
         }
 
         await super.deleteUser(id);
-        if (errors) {
+        if (errors.length > 0) {
```

Comment:
@JanKoehnlein, actually, after seeing more instances of this issue, I belief it's not a good idea in general to throw an error after `super.deleteUser` is actually done.
This leads to a state, where one's account is locked in a `deleted` state. This definitely requires proper handling in the frontend.

Reply:
I was mostly worried about users that have a GitHub-based subscription. They actually cannot cancel that subscription using our service, so they should get a message. Similar when Chargebee fails any reason. In any case, users wanted to delete their account so I thought it would be OK if that is the state on the Gitpod side. 

But I am all in for proper error propagation to the frontend. Just unsure which actions should be taken/rolled back on the Gitpod side if some of the error conditions are met.

Reply:
That's not an easy issue to solve. While error propagation isn't the real adventure, to decide on the loose end is, especially because the account data will be deleted. 
Maybe we should not call `super.deleteUser` if there are issues to cancel subscription. AND propagate an error.